File: working_with_user_events/silverlight_80/charged_event.cs
Dictionary<string, object> chargeDetails = new Dictionary<string, object>();
chargeDetails.Add("Amount", 300);
chargeDetails.Add("Currency", "USD");
chargeDetails.Add("Payment mode", "Credit card");
chargeDetails.Add("Charged ID", 24052013); // important to avoid duplicate transactions due to network failure

Dictionary<string, object> item1 = new Dictionary<string, object>();
item1.Add("Category", "books");
item1.Add("Book name","The Millionaire next door");
item1.Add("Quantity", 1);

Dictionary<string, object> item2 = new Dictionary<string, object>();
item2.Add("Category", "books");
item2.Add("Book name","Achieving inner zen");
item2.Add("Quantity", 1);

Dictionary<string, object> item3 = new Dictionary<string, object>();
item3.Add("Category", "books");
item3.Add("Book name","Chuck it, let's do it");
item3.Add("Quantity", 5);

List<Dictionary<string, object>> items = new List<Dictionary<string, object>>();
items.Add(item1);
items.Add(item2);
items.Add(item3);

CleverTapInstance.Event.PushCharged(chargeDetails, items);

File: working_with_user_events/android/charged_event.java
HashMap<String, Object> chargeDetails = new HashMap<String, Object>();
chargeDetails.put("Amount", 300);
chargeDetails.put("Payment Mode", "Credit card");
chargeDetails.put("Charged ID", 24052013); // important to avoid duplicate transactions due to network failure

HashMap<String, Object> item1 = new HashMap<String, Object>();
item1.put("Product category", "books");
item1.put("Book name", "The Millionaire next door");
item1.put("Quantity", 1);

HashMap<String, Object> item2 = new HashMap<String, Object>();
item2.put("Product category", "books");
item2.put("Book name", "Achieving inner zen");
item2.put("Quantity", 1);

HashMap<String, Object> item3 = new HashMap<String, Object>();
item3.put("Product category", "books");
item3.put("Book name", "Chuck it, let's do it");
item3.put("Quantity", 5);

ArrayList<HashMap<String, Object>> items = new ArrayList<HashMap<String, Object>>();
items.add(item1);
items.add(item2);
items.add(item3);

try {
    cleverTap.event.push(CleverTapAPI.CHARGED_EVENT, chargeDetails, items);
} catch (InvalidEventNameException e) {
    // You have to specify the first parameter to push()
    // as CleverTapAPI.CHARGED_EVENT
}
